src/Compilers/CSharp/Test/Semantic/Semantics/NullCheckedParameterTests.cs

```diff
@@ -170,5 +170,295 @@ class C
                     //     public static extern int M(int x!);
                     Diagnostic(ErrorCode.ERR_MustNullCheckInImplementation, "x").WithArguments("x").WithLocation(6, 36));
         }
+
+        [Fact]
+        public void NullCheckedMethodDeclaration()
+        {
+            var source = @"
+class C
+{
+    void M(string name!) { }
+}";
+            CreateCompilation(source).VerifyDiagnostics();
```

Comment:
These are good tests, but we should verify that they work properly in the API by calling something like:

```C#
var m = comp.GlobalNamespace.GetMember("C").GetMember("M);
Assert.True(m.Parameters[0].IsNullChecked);
```

and there should also be a negative case, so verify that a similar method `M2` without the `!` returns false in the API

Reply:
Pushed but will change Assert Equals toAssert True